### ISSUE TITLE
Button continues to display loading animation after error

### ISSUE THREAD

**Issue by karlfock** [Adyen/adyen-web#1485]
**Describe the bug**
This first happens in version 5.6.2. When entering card details with a card number that is non 3DS and entering an incorrect CVC number, then there is no redirect and the error response is handled directly by the page which contains the Adyen dropin.

In this case the Pay button continues loading even though the payment request has terminated and there was an error response sent back. 

**To Reproduce**
1. Go to page containing dropin
2. Enter card number (non 3DS test card number): 2222 4000 7000 0005 
3. Enter expire date: 03/30
4. Enter incorrect CVC, for example: 111

**Expected behavior**
When the payment request has been sent and there is a response the loading animation on the Pay button should disappear and it should be possible to click Pay again. This is not possible, you have to reload the page. 

**Screenshots**
![image](https://user-images.githubusercontent.com/2572611/153173306-51953c12-7bb4-4281-968a-91e8f730f765.png)

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.1
 - Browser: Firefox and Chrome
 - Version Firefox 96.0.3 / Chrome 97.0

**Comment by ribeiroguilherme** [comment]
Hi @karlfock ,

Thanks for reaching out. Are you experiencing this only on v5.6.2? Have you tried other versions too?

I tried to reproduce the problem locally but didn't manage to. Can you show how you are creating the drop-in and its event handlers that you attached?

**Comment by karlfock** [comment]
I just paste in the entire source code file that we use to render the dropin:

```
import AdyenCheckout from '@adyen/adyen-web';
import React from 'react';
import { IntlShape } from 'react-intl';
import DropinElement from '@adyen/adyen-web/dist/types/components/Dropin';
import { extractSearchParams } from '@hi3g/utils';
import {
  PaymentState,
  PaymentRequest,
  AdditionalDetailsState,
  PaymentDetailsRequest,
  AdyenCheckoutConfiguration,
  CheckoutConfig,
} from '../types';
import { CHANNEL, ERROR_SCOPE } from '../constants/constants';
import messages from '../global.messages';
import logger from './logger';
import filterPaymentMethods from './filterPaymentMethods';

type Params = {
  clientOrderId: string;
  checkoutConfig: CheckoutConfig;
  enableStoreDetails: boolean;
  formatMessage: IntlShape['formatMessage'];
  paymentContainer: React.MutableRefObject<HTMLDivElement>;
  showPrice: boolean;
  submitPaymentCallback: (request: PaymentRequest) => void;
  submitPaymentDetailsCallback: (
    paymentDetailsRequest: PaymentDetailsRequest,
  ) => void;
  returnUrl?: string;
};

const isBrowser = typeof window !== 'undefined';

const { showHolderName = '' } = isBrowser
  ? extractSearchParams(window.location.href)
  : {};

const getHasHolderName = (): boolean => {
  if (process.env.GATSBY_TARGET_ENV !== 'production') {
    return showHolderName === 'true';
  }
  return false;
};

const createAdyenDropin = async ({
  clientOrderId,
  checkoutConfig,
  enableStoreDetails,
  formatMessage,
  paymentContainer,
  showPrice,
  submitPaymentCallback,
  submitPaymentDetailsCallback,
  returnUrl,
}: Params): Promise<DropinElement | null> => {
  const {
    currency,
    customerSelectedPaymentMethod,
    customerSelectedSignupRecurring,
    dropinData: { clientKey, environment, locale, paymentMethods },
    paymentId,
    totalPrice,
  } = checkoutConfig;
  try {
    const configuration = {
      clientKey,
      environment: environment || 'test',
      locale: locale || 'da_DK',
      onAdditionalDetails: (state: AdditionalDetailsState) => {
        submitPaymentDetailsCallback(state.data);
      },
      onSubmit: (state: PaymentState) => {
        if (state.isValid) {
          const data = {
            amount: {
              currency,
              value: totalPrice,
            },
            clientOrderId,
            dropinData: {
              ...state.data,
              channel: CHANNEL,
              origin: window.location.origin,
            },
            paymentId,
            returnUrl,
          };
          submitPaymentCallback(data);
        }
      },
      paymentMethodsConfiguration: {
        card: {
          enableStoreDetails,
          hasHolderName: getHasHolderName(),
          name: formatMessage(messages.creditCard),
        },
        mobilepay: {
          name: 'MobilePay',
        },
      },
      paymentMethodsResponse: {
        paymentMethods: filterPaymentMethods(
          paymentMethods,
          customerSelectedSignupRecurring,
          customerSelectedPaymentMethod,
        ),
      },
      translations: {
        da_DK: {
          payButton: formatMessage(
            showPrice ? messages.pay : messages.saveCardInfo,
          ),
          storeDetails: formatMessage(messages.recurringPayment),
        },
      },
    } as AdyenCheckoutConfiguration;

    const checkout = await AdyenCheckout(configuration);
    return checkout.create('dropin').mount(paymentContainer.current);
  } catch (error) {
    logger.error({
      component: 'dropinCheckout',
      error,
      message: 'Error in dropinCheckout',
      scope: ERROR_SCOPE,
    });
    return null;
  }
};

export default createAdyenDropin;

```

**Comment by ribeiroguilherme** [comment]
Hi @karlfock ,

There was an issue with how drop-in was handling the `status`  which is fixed on #1394 , and released on v5.6.2. I believe that is why you noticed this behavior.

You should handle the 'status' by yourself when you get the result from the payments call - that is the intended behavior. Drop-in will set the status to 'loading' by default when 'Pay' button is pressed, but the next step is up to you according to the result got from the payment call. 

You can see that being done by the `showFinalResult` function on the code snippet of [this section](https://docs.adyen.com/online-payments/web-drop-in/integrated-before-5-0-0#step-2-add-drop-in) on our docs.

From your example, I believe that the following code should make it work:

```
onSubmit: async (state: PaymentState, dropin) => {
    if (state.isValid) {
        const data = { ... }
        const result = await submitPaymentCallback(data);
           
        if (result.resultCode === 'Refused') {
            dropin.setStatus('ready');
            // show error
        } else {
            // handle action, success, whatever..
        }
    }
},
```

You can also check the [onSubmit function on our dev playground](https://github.com/Adyen/adyen-web/blob/master/packages/playground/src/pages/Dropin/manual.js#L22) which handles the status of the component by checking the result of the payment call.  

Let me know if that fixes your problem.

**Comment by karlfock** [comment]
Thank you very much @ribeiroguilherme, that fixes it!
